.github/workflows/publish-npm.yml

```diff
@@ -14,17 +17,22 @@ jobs:
         uses: actions/checkout@v2
         with:
           ref: main
+          persist-credentials: false
       # limit releases to version changes - https://github.com/EndBug/version-check
+      # EndBug/version-check@v1 as of Oct 8, 2020
       - name: Check version changes
-        uses: EndBug/version-check@v1
+        uses: EndBug/version-check@eea9dab124be9214bd32356f32f7d8945173db13
         id: version_check
         with:
           # diff the commits rather than commit message for version changes
           diff-search: true
 
       - name: Version update detected
         if: steps.version_check.outputs.changed == 'true'
-        run: 'echo "Version change found! New version: ${{ steps.version_check.outputs.version }} (${{ steps.version_check.outputs.type }})"'
+        run: |
+          printf "Version change found! New version: '%s' (%s)\n" \
+            "${{ steps.version_check.outputs.version }}" \
+            "${{ steps.version_check.outputs.type }}"
```

Comment:
```suggestion
        env:
          VERSION: ${{ steps.version_check.outputs.version }}
          TYPE: ${{ steps.version_check.outputs.type }}
        run: |
          printf "Version change found! New version: '%s' (%s)\n" \
            "$VERSION" \
            "$TYPE"
```

Freestyling here, but this should be gtg

Reply:
updated to use `env:`
![Screenshot 2025-04-30 at 9 38 06 AM](https://github.com/user-attachments/assets/1717b984-7f8b-4f29-b742-a4431d025a22)